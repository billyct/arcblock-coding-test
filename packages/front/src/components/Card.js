import PropTypes from 'prop-types'
import cn from 'classnames'

const Card = ({title, children, className}) => (
  <div className='rounded shadow  py-6 bg-white mb-4'>
    <h3 className='text-xl mb-4 px-6'>
      {title}
    </h3>
    <div className={cn('px-6', className)}>
      {children}
    </div>
  </div>
)

Card.propTypes = {
  title: PropTypes.string.isRequired,
}

export default Card
