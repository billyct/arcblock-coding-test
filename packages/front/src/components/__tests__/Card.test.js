import {render} from '@testing-library/react'

import Card from '../Card'

describe.each([
  [
    {
    title: 'title',
    className: 'classname',
    children: 'children',
    }
  ],
  [
    {title: 'title'}
  ],
])('test Card Component with %o', attribute => {
  it('should render correct', function () {
    const {container} = render(<Card {...attribute}/>)
    expect(container).toMatchSnapshot()
  })
})

