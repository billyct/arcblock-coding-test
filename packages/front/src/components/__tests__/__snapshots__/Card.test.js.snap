// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Card Component with { title: 'title' } should render correct 1`] = `
<div>
  <div
    class="rounded shadow  py-6 bg-white mb-4"
  >
    <h3
      class="text-xl mb-4 px-6"
    >
      title
    </h3>
    <div
      class="px-6"
    />
  </div>
</div>
`;

exports[`test Card Component with { title: 'title', className: 'classname', children: 'children' } should render correct 1`] = `
<div>
  <div
    class="rounded shadow  py-6 bg-white mb-4"
  >
    <h3
      class="text-xl mb-4 px-6"
    >
      title
    </h3>
    <div
      class="px-6 classname"
    >
      children
    </div>
  </div>
</div>
`;
